src/Symfony/Component/Scheduler/Generator/MessageGenerator.php

```diff
@@ -137,4 +159,13 @@ private function checkpoint(): Checkpoint
     {
         return $this->checkpoint ??= new Checkpoint('scheduler_checkpoint_'.$this->name, $this->getSchedule()->getLock(), $this->getSchedule()->getState());
     }
+
+    private function getMessageIdAndRun(object $message, MessageContext $contest): string
+    {
+        return match (true) {
+            $message instanceof ServiceCallMessage => $message->__toString().json_encode($message->getArguments()),
+            $message instanceof \Stringable => $message->__toString(),
+            default => $message::class,
+        }.$contest->triggeredAt->format('_YmdHis');
```

Comment:
I think you can use MessageContext::$id instead

```suggestion
        return $contest->id.$contest->triggeredAt->format('_YmdHis');
```